[MISC] deprecate seqan3::output_file_validator constructor (#2460)
seqan/seqan3: CHANGELOG.md

#### Argument Parser

> * `seqan3::output_file_validator` can't be constructed with the extension list alone any more, you need to specify one

Reviewer: ```suggestion
* `seqan3::output_file_validator` cannot be constructed with the extension list alone anymore, you need to specify one
```
